### PR TITLE
mehtod find_file parenthesis added

### PR DESCRIPTION
**Pull request by AliceLanniste**
when I'm trying to add testcases for `find_file`,I found a question that `f.largest` and `ikey`can't be compared, type of `f.largest` is `InternalKey` ,and  type of `ikey` is `Slice`,though the returned output of the two is `&[u8]`,I think the comparation of the two type is wrong, I read the `Leveldb`source code, the ikey of test of‘Findfile’ is encoded `InternalKey` to `Slice`,but in `Wickdb`, I didn't similar method or function

### CODE REVIEW

**Comment by AliceLanniste** [comment]
hi, I founed a bug when  I  using `cargo test -- all`run testcases for `Version::find_file`.  Sometimes all passed ,sometimes failed. The outputs are different for every test.   I added two testcase in `version/mod.rs`,one named`test_find_file`,another`test_find_file2`.

there are two outputs when I run finally
### first output of ` cargo test`

```
test version::tests::test_find_file ... ok
test version::tests::test_find_files2 ... ok
test version::version_edit::tests::test_encode_decode ... ok
test version::version_edit::tests::test_set_comparator_name ... ok
test version::version_edit::tests::test_set_last_sequence ... ok
test version::version_edit::tests::test_set_log_number ... ok
test version::version_edit::tests::test_set_next_file ... ok
test version::version_edit::tests::test_set_prev_log_number ... ok
test util::coding::tests::test_put_fixed32 ... ok
test record::tests::test_many_blocks ... ok
test mem::arena::tests::test_allocate_mixed ... ok
test record::tests::test_random_read ... ok
test mem::skiplist::tests::test_concurrency_read_write ... ok
test sstable::tests::test_randomized_key ... ok

test result: ok. 142 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests wickdb

```
### second output 
```
test version::tests::test_find_file ... ok
test version::tests::test_find_files2 ... FAILED
test version::version_edit::tests::test_encode_decode ... ok
test version::version_edit::tests::test_set_comparator_name ... ok
test version::version_edit::tests::test_set_log_number ... ok
test version::version_edit::tests::test_set_next_file ... ok
test version::version_edit::tests::test_set_prev_log_number ... ok
test version::version_edit::tests::test_set_last_sequence ... ok
test util::coding::tests::test_put_fixed32 ... ok
test record::tests::test_many_blocks ... ok
test mem::arena::tests::test_allocate_mixed ... ok
test mem::skiplist::tests::test_concurrency_read_write ... ok
test record::tests::test_random_read ... ok
test sstable::tests::test_randomized_key ... ok

failures:

---- version::tests::test_find_files2 stdout ----
thread 'version::tests::test_find_files2' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `0`', src\version\mod.rs:700:9


failures:
    version::tests::test_find_files2

test result: FAILED. 141 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

```
the error is `assert_eq!(2, file_metas.find(&Slice::from("301")));` 
 It‘s impossible, if you read my testcase code

**Comment by Fullstop000** [comment]
Thank you for your contribution!. I'm investigating this. Do you mind me modifying your PR?

**Comment by AliceLanniste** [comment]
yesterday I fixed `file.number` error.but I didn't  update.  Please wait for me to submit a new pr

**Comment by AliceLanniste** [comment]
I'm sorry, I forgot I've updated. you can modify my PR

**Comment by Fullstop000** [comment]
@AliceLanniste I make some modifications and now the test cases are stable. PTAL

**Comment by AliceLanniste** [comment]
Sorry,I'll pay attention. I wonder why  added `let test_case...`can pass successfully

File: src/version/mod.rs [hunk 1]
```diff
@@ -637,3 +637,87 @@ impl Iterator for LevelFileNumIterator {
...
+        fn generate(&mut self, smallest: &Slice, largest: &Slice) {
+            let mut file = FileMetaData::default();
+            file.number = self.files.len() as u64 + 1;
+            file.smallest = Rc::new(InternalKey::new(smallest, 100, ValueType::Value));
+            file.largest = Rc::new(InternalKey::new(largest, 100, ValueType::Value));
+            self.files.push(Arc::new(file));
+        }
+
+        fn find(&self, key: &Slice) -> usize {
+            let target = Slice::from(InternalKey::new(key, 100, ValueType::Value).data());
```
Comment (Fullstop000) [inline src/version/mod.rs]: The `InternalKey::new(key, 100, ValueType::Value).data()` doesn't  be alive as long as the outer `Slice` which will make `target` become a UB.

Comment (Fullstop000) [reply src/version/mod.rs]: @AliceLanniste Finally I found this 😂

Comment (Fullstop000) [inline src/version/mod.rs]:
```suggestion
            let ikey = InternalKey::new(key, 100, ValueType::Value);
            let target = Slice::from(ikey.data());
```

**Comment by Fullstop000** [comment]
@AliceLanniste Do you have any contact information (like Email) so that we can keep in touch? We can just use wechat.

**Comment by AliceLanniste** [comment]
My weChat account is registered very early. WeChat account is very long and complicated. 

You can find me on WeChat through my&nbsp; email number 1399789151.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Fullstop000"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月11日(星期三) 下午5:13
收件人:&nbsp;"Fullstop000/wickdb"<wickdb@noreply.github.com&gt;;
抄送:&nbsp;"酷拉皮卡"<1399789151@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [Fullstop000/wickdb] mehtod find_file parenthesis added (#22)




@AliceLanniste Do you have any contact information (like Email) so that we can keep in touch? We can just use wechat.

**Comment by Fullstop000** [comment]
> email number 1399789151.

@AliceLanniste Is that a QQ number or something else?

**Comment by AliceLanniste** [comment]
> 
> 
> > email number 1399789151.
> 
> @AliceLanniste Is that a QQ number or something else?

yes that is a QQ number

**Pull request merged**